src/test/java/uk/gov/moj/sdt/service/FeedbackServiceTest.java

```diff
@@ -0,0 +1,151 @@
+package uk.gov.moj.sdt.service;
+
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.ExtendWith;
+import org.mockito.Mock;
+import org.mockito.junit.jupiter.MockitoExtension;
+import uk.gov.moj.sdt.cmc.consumers.converter.XmlConverter;
+import uk.gov.moj.sdt.cmc.consumers.response.JudgmentWarrantStatus;
+import uk.gov.moj.sdt.cmc.consumers.response.ProcessingStatus;
+import uk.gov.moj.sdt.dao.api.IIndividualRequestDao;
+import uk.gov.moj.sdt.domain.api.IErrorLog;
+import uk.gov.moj.sdt.domain.api.IIndividualRequest;
+import uk.gov.moj.sdt.request.CMCFeedback;
+import uk.gov.moj.sdt.utils.cmc.RequestType;
+
+import java.io.IOException;
+import java.util.Calendar;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.Mockito.mock;
+import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.when;
+
+@ExtendWith(MockitoExtension.class)
+class FeedbackServiceTest {
+
+    @Mock
+    IIndividualRequestDao individualRequestDao;
+
+    @Mock
+    XmlConverter xmlToObject;
+
+    private FeedbackService feedbackService;
+
+    private static final String SDT_REQUEST_ID = "MCOL-10012013010101-100099999";
+
+    private String CLAIM_ERROR = "Claim Error";
+
+    private String CLAIM_NUMBER = "123456";
+
+    private String WARRANT_NUMBER = "123457";
+
+    private String ENFORCING_COURT_NAME = "London Court";
+
+    private String ENFORCING_COURT_CODE = "COURT-1234";
+
+    private int ERROR_CODE = 1234;
+
+    @BeforeEach
+    public void setUp() {
+        feedbackService = new FeedbackService(individualRequestDao, xmlToObject);
+    }
+
+    @Test
+    void shouldPersistClaim() throws IOException {
+        IIndividualRequest individualRequest = mock(IIndividualRequest.class);
```

Comment:
To make the test more comprehensive, the IndividualRequest object could be changed to an actual object rather than a mock object and checks made of the fields on it to ensure that they've been set to the values from the CMCFeedback object.

For example:
```java
IIndividualRequest individualRequest = new IndividualRequest();
...
feedbackService.feedback(SDT_REQUEST_ID, cmcFeedback);
assertEquals(ERROR_CODE, individualRequest.getErrorLog().getErrorCode);
```

Reply:
I don't think this is needed, but will do the changes

Reply:
done now

Reply:
Could you add the following JUnit assert checks?

**shouldPersistClaim()**
Check errorText, claimNumber, issueDate and serviceDate values on individualRequest.

**shouldPersistJudgment()**
Check errorText, judgmentEnteredDate and firstPaymentDate values on individualRequest.

**shouldPersistBreathingSpace()**
Check errorText on individualRequest.

**shouldPersistClaimStatusUpdate()**
Check errorText on individualRequest.

**shouldPersistWarrant()**
Check fee, warrantNumber, enforcingCourtCode, enforcingCourtName and that errorLog hasn't been set on individualRequest.

**shouldPersistJudgementWarrant()**
Check judgmentWarrantStatus, fee, warrantNumber, enforcingCourtCode, enforcingCourtName, judgmentEnteredDate, firstPaymentDate, issueDate and that errorLog hasn't been set on individualRequest.

Reply:
addressed